crates/holochain/src/conductor/conductor.rs

```diff
@@ -1112,26 +1120,29 @@ where
     /// Update the internal state with a pure function mapping old state to new,
     /// which may also produce an output value which will be the output of
     /// this function
-    async fn update_state_prime<F: Send, O>(&self, f: F) -> ConductorResult<(ConductorState, O)>
+    async fn update_state_prime<F, O>(&self, f: F) -> ConductorResult<(ConductorState, O)>
     where
         F: FnOnce(ConductorState) -> ConductorResult<(ConductorState, O)>,
+        O: Send,
     {
         self.check_running()?;
-        let mut guard = self.env.conn()?;
-        let output = guard.with_commit(|txn| {
-            let state = txn
-                .query_row("SELECT blob FROM ConductorState WHERE id = 1", [], |row| {
-                    row.get("blob")
-                })
-                .optional()?;
-            let state = match state {
-                Some(state) => from_blob(state)?,
-                None => ConductorState::default(),
-            };
-            let (new_state, output) = f(state)?;
-            mutations::insert_conductor_state(txn, (&new_state).try_into()?)?;
-            Result::<_, ConductorError>::Ok((new_state, output))
-        })?;
+        let output = self
+            .env
+            .async_commit_in_place(move |txn| {
```

Comment:
I'm sure you ran into something that made this not `'static` but I didn't notice what it was in my cursory look around at callers... do you think we should put a deprecated or something on the `async_commit_in_place` definition as a hint we should try to remove it?

Reply:
It's the DnaStore which can't be cloned. So we'd have to put the DnaStore inside a `Arc<Mutex<DnaStore>>` if we want to clone it into this.